### PR TITLE
Order locations logs search alphabetically 

### PR DESCRIPTION
**Pull request by georgeschena**
Screenshot 2019-03-19 at 12 08 45

### CODE REVIEW

File: app/views/logs_searches/location.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <%= form_for @search do |f| %>
       <div class='govuk-form-group'>
         <%= f.label :term, "Select one of your organisation's locations", class: 'govuk-label' %>
-        <%= collection_select :logs_search, :term, current_organisation.locations, :id, :full_address, {}, { class: 'govuk-select' } %>
+        <%= collection_select :logs_search, :term, current_organisation.locations.order([:address]), :id, :full_address, {}, { class: 'govuk-select' } %>
```
Comment (tommotaylor) [inline app/views/logs_searches/location.html.erb]: I would suggest putting this into the controller so the view just uses an instance variable such as `@locations`

File: spec/features/logging/view_auth_requests_for_a_location_spec.rb [hunk 2]
```diff
@@ -11,6 +11,19 @@
...
+
+      let!(:organisation) { create(:organisation, locations: [location_2, location_3, location_4]) }
+
+      it 'shows me the locations in alpabetical order' do
+        locations = find('#logs_search_term').all('option').collect(&:text)
+        expect(locations).to match_array ["Abbey Street, HA7 2BL", "Garry Road, HA7 4BL", "Zeon Grove, HA7 3BL"]
+      end
+    end
+
     it 'shows me my locations' do
```
Comment (tommotaylor) [inline spec/features/logging/view_auth_requests_for_a_location_spec.rb]: This spec seems redundant now, the above spec tests for all the locations.

File: spec/features/logging/view_auth_requests_for_a_location_spec.rb [hunk 3]
```diff
@@ -11,6 +11,19 @@
       click_on 'Go to search'
     end
 
+    context 'With a new set of locations' do
```
Comment (tommotaylor) [inline spec/features/logging/view_auth_requests_for_a_location_spec.rb]: Not a very descriptive context. Doesn't fit in with the rest of the spec. I wonder if we need this context at all.

File: app/controllers/logs_searches_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class LogsSearchesController < ApplicationController
   def create
     @search = LogsSearch.new(search_params)
+    @locations = current_organisation.locations.order([:address])
```
Comment (cybojenix) [inline app/controllers/logs_searches_controller.rb]: how come this isn't in the `location` controller?

Comment (georgeschena) [reply app/controllers/logs_searches_controller.rb]: Because we are using the LogsSearch controller to retrieve the logs for each filter

Comment (georgeschena) [reply app/controllers/logs_searches_controller.rb]: I do feel this LogsSearchController is a tad confusing imo

Comment (tommotaylor) [reply app/controllers/logs_searches_controller.rb]: Looks like its in the wrong controller action but doesn't work the other way. I'm also a little confused about how this controller is setup and how the routes work.

**Pull request merged**